app/controllers/works_controller.rb

```diff
@@ -0,0 +1,84 @@
+class WorksController < ApplicationController
+
+  def index
+    @works = Work.all
+    @movies = Work.where(category: 'movie')
+    @books = Work.where(category: 'book')
+    @albums = Work.where(category: 'album')
+  end
+
+  def show
+    @work = Work.find_by(id: params[:id])
```

Comment:
This would be a great opportunity for a controller filter as this exact code is used in multiple methods